p2p/src/authenticated/actors/tracker/peer_set.rs

```diff
@@ -0,0 +1,69 @@
+use crate::authenticated::types;
+use bitvec::{order::Lsb0, vec::BitVec};
+use commonware_utils::Array;
+use std::collections::HashMap;
+
+/// Represents a set of peers and their knowledge of each other.
+pub struct PeerSet<P: Array> {
+    /// The index at which this peer set applies.
+    pub index: u64,
+
+    /// The list of peers, sorted.
+    pub sorted: Vec<P>,
+
+    /// The index of each peer in the sorted list, for quick lookup.
+    pub order: HashMap<P, usize>,
+
+    /// My knowledge of each peer in the set.
+    pub knowledge: BitVec<u8, Lsb0>,
+
+    /// The message to send to other peers.
+    pub msg: types::BitVec,
+}
+
+impl<P: Array> PeerSet<P> {
+    pub fn new(index: u64, mut peers: Vec<P>) -> Self {
+        // Insert peers in sorted order
+        peers.sort();
+        let mut order = HashMap::new();
+        for (idx, peer) in peers.iter().enumerate() {
+            order.insert(peer.clone(), idx);
+        }
+
+        // Create bit vector
+        let knowledge = BitVec::repeat(false, peers.len());
```

Comment:
If I'm not mistaken, this could still result in bits beyond `peers.len()` to be non-zero when encoded as a vec below: https://docs.rs/bitvec/latest/bitvec/vec/struct.BitVec.html#method.into_vec

I think we may need to call `set_uninitialized(false)` after this? https://docs.rs/bitvec/latest/bitvec/vec/struct.BitVec.html#method.set_uninitialized

Reply:
Or just replace this with #641

Reply:
Yeah so the code here is not new, this is how the logic is already. I did identify this issue with the code, however since the size of the bitvec does not change (the number of peers is constant per index), there should not be any non-zero bits beyond `peers.len()`. I believe that can only occur if you modify the size of the bitvec.

We do check that the items past the length of the bitvec are zeroed on deserialization, so yes there would be an issue if these weren't zeroed. But my plan was to just add the `set_uninitialized` logic (or manually do it on the last byte) during #641 . I will reference this comment chain in that issue

Reply:
Yeah, this was definitely my mess up lol.

I think we should call set_unitialized() for now and plan on getting on to more stable ground later.